packages/typespec-ts/test/modularUnit/operations.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { assert } from "chai";
+import { emitModularOperationsFromTypeSpec } from "../util/emitUtil.js";
+import { assertEqualContent } from "../util/testUtil.js";
+import { Diagnostic } from "@typespec/compiler";
+
+describe("operations", () => {
+  it("required & optional & nullable headers", async () => {
+    const tspContent = `
+      model Bar {
+        prop1: string;
+        prop2: int64;
+      }
+      @encode(BytesKnownEncoding.base64url)
+      scalar base64urlBytes extends bytes;
+      op read(
+        @header requiredHeader: string, 
+        @header optionalHeader?: string,  
+        @header nullableOptionalHeader?: string | null,
+        @header bytesHeader: bytes,
+        @header @encode(BytesKnownEncoding.base64) value: bytes,
+        @header({
+          format: "csv",
+        })
+        csvArrayHeader: base64urlBytes[],
+        @header utcDateHeader: utcDateTime,
+        @header optionalDateHeader?: utcDateTime,
+        @header nullableDateHeader?: utcDateTime | null,
+        ...Bar): OkResponse;
+        `;
+
+    const operationFiles = await emitModularOperationsFromTypeSpec(tspContent);
+    assert.ok(operationFiles);
+    assert.equal(operationFiles?.length, 1);
+    // console.log(operationFiles?.[0]?.getFullText()!);
+    assertEqualContent(
+      operationFiles?.[0]?.getFullText()!,
+      `
+      import { TestingContext as Client } from "../rest/index.js";
+      import { StreamableMethod, operationOptionsToRequestParameters } from "@azure-rest/core-client";
+      import { uint8ArrayToString } from "@azure/core-util";
+      
+      export function _readSend(
+        context: Client, 
+        requiredHeader: string,
+        bytesHeader: Uint8Array,
+        value: Uint8Array,
+        csvArrayHeader: Uint8Array[],
+        utcDateHeader: Date, 
+        body: Bar, 
+        options: ReadOptions = { requestOptions: {} }): StreamableMethod<Read200Response> {
+          return context.path("/", ).post({...operationOptionsToRequestParameters(options), 
+            headers: {
+              "required-header": requiredHeader,
+              ...(options?.optionalHeader !== undefined
```

Comment:
Is it complicate to apply this to build body parameter logic ?

Reply:
Yeah, I would not recommand to do this in body, it would cause verbose code.